osu.Game/Screens/Spectate/SpectatorScreen.cs

```diff
@@ -26,7 +27,8 @@ namespace osu.Game.Screens.Spectate
     /// </summary>
     public abstract class SpectatorScreen : OsuScreen
     {
-        private readonly int[] userIds;
+        protected IEnumerable<int> GetUserIds() => userIds;
+        private readonly HashSet<int> userIds = new HashSet<int>();
```

Comment:
Rather than the weird `GetUserIds()` method, are you okay with making this a `List<>` and exposing as a `protected IReadOnlyList<>` for use in `MultiplayerSpectatorScreen`?

Reply:
I've made this change anyways, see if you agree with it.